pipeline/lib/storage.ml

```diff
@@ -0,0 +1,81 @@
+let with_db ~conninfo fn =
+  let db = new Postgresql.connection ~conninfo () in
+  let ret_val = fn db in
+  db#finish;
+  ret_val
+
+let setup_metadata ~repository (db : Postgresql.connection) =
+  Logs.debug (fun log -> log "Inserting metada....");
+  let run_at = Sql_util.time (Ptime_clock.now ()) in
+  let repo_id = Sql_util.string (Repository.info repository) in
+  let commit = Sql_util.string (Repository.commit_hash repository) in
+  let branch = Sql_util.(option string) (Repository.branch repository) in
+  let pull_number = Sql_util.(option int) (Repository.pull_number repository) in
+  let query =
+    (*
+      When setting up metadata, we are only insert the details that we know at th
+      beginning. If we see a conflict here, that means we have started running the
+      benchmarks again for this repo and commit, so we reset the build_job_id
+      and the run_job_id.
+    *)
+    Fmt.str
+      {|
+    INSERT INTO
+    benchmark_metadata(run_at, repo_id, commit, branch, pull_number)
+    VALUES
+    (%s, %s, %s, %s, %s)
+    ON CONFLICT(repo_id, commit) DO UPDATE
```

Comment:
For this `ON CONFLICT` condition to be triggered, we'd need to add a UNIQUE constraint for `(repo_id, commit)` on the table.

Reply:
Yes, nice catch @punchagan :) Thanks!